### DISCUSSION TITLE
Submit 時出現 Runtime Error

### DISCUSSION THREAD

**Discussion by weih210287** [Lidemy/forum#85] [其他問題] [answered]
### Q1. 請問如果 Submit 時出現 Runtime Error 的話怎麼辦？
我自己上網查的結果是程式跑到一半崩潰了或是哪一行輸入有問題，例如我在解 LIOJ 1017 貪婪的小偷時在自己的電腦上測試沒問題可是放上 LIOJ 就會出現 Runtime Error。自己測試的一番後以及各種 console.log，猜測原因可能出在 `let tmp = lines.split('')` 這邊，因為當我把它改成 `let tmp = lines[0].split('')` 多加位數後就會變成 WA，雖然不對但比起跑不動至少是可行的，可是怎麼改都還是不對，最後是換個方法把所有 price 加進新的數列才成功解題。

我原本的程式碼如下：
```javascript
function solve(lines) {
  let c = Number(lines[0])
  let n = Number(lines[1])
  let tmp = lines.split('')  // 轉成數列
  tmp.splice(0, 2)  // 去掉 c & n 只剩所有物品的 price
  for(let j = 0; j < tmp.length; j++) {  // 把 price 按大到小排序
    for(let i = 0; i < tmp.length; i++) {
      if((tmp[i]) < (tmp[i+1])) {
        [(tmp[i]), (tmp[i+1])] = [(tmp[i+1]), (tmp[i])]
      }
    }
  }  

  var max = 0;
  if(c > n) {
    for(let p = 0; p < tmp.length; p++) {  // 如果 c > n，加總所有 price
      max += Number(lines[p])
    }  
  } else {
    for(let l = 0; l < c; l++) {  // 加總前 c 大的 price
      max += Number(tmp[l]) 
    }
  }  
  console.log(max)
}
```

謝謝大家 qwq !!

**Comment by EshauHuang** [comment, marked as answer]
你可以試著把 lines 印出來看看，你就會知道會什麼需要這樣改了!

簡單來說就是 split 是拿來處理「字串」，而你的 lines 是有很多筆字串資料的「陣列」。
所以你必須先提取陣列的元素後才能使用 split 來分割。

**Reply by weih210287** [reply]
啊啊了解，謝謝你 ˊ ˇ ˋ 
沒有想到這個方法，之前一直以為 lines 輸進去的是字串 😅
剛剛也看到了 MTR05 的更新影片

**Reply by EshauHuang** [reply]
你輸進去的是字串沒錯喔，只是依照你輸入的第一行、第二行......，存成矩陣的第一個、第二個......元素。

**Reply by weih210287** [reply]
原來如此!!
真的非常感謝，我再去把 LIOJ 的轉換教學看一次 TvT
